plugins/spacelift/api_key.go

```diff
@@ -0,0 +1,101 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://docs.spacelift.io/integrations/api#spacelift-api-key-token"),
+		ManagementURL: sdk.URL("https://mycorp.app.spacelift.io/settings/api-keys"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Endpoint,
+				MarkdownDescription: "The URL to your Spacelift account, for example https://mycorp.app.spacelift.io",
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+					Prefix: "https://",
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "The ID of your Spacelift API key. Available via the Spacelift application.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Uppercase: true,
+						Digits:    true,
+					},
+				},
+			},
+			{
+				Name:                fieldname.APIKeySecret,
+				MarkdownDescription: "The secret for your API key. Only available when the secret is created.",
+				Secret:              true,
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+					},
+				},
+			},
+		},
+		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
+		Importer: importer.TryAll(
+			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TrySpaceliftConfigFile(),
+		)}
+}
+
+var defaultEnvVarMapping = map[string]sdk.FieldName{
+	"SPACELIFT_API_KEY_ENDPOINT": fieldname.Endpoint,
+	"SPACELIFT_API_KEY_ID": fieldname.APIKeyID,
+	"SPACELIFT_API_KEY_SECRET": fieldname.APIKeySecret,
+}
+
+func TrySpaceliftConfigFile() sdk.Importer {
+	return importer.TryFile("~/.spacelift/config.json", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		var config Config
+		if err := contents.ToJSON(&config); err != nil {
+			out.AddError(err)
+			return
+		}
+
+		for name, profile := range config.Profiles {
+			if profile.Credentials.Endpoint == "" {
+				continue
+			}
+
+			out.AddCandidate(sdk.ImportCandidate{
```

Comment:
Is there a reason why we are not importing the KeyId and KeySecret? Ideally, we'd want them stored in the 1Password item as well without any credentials left on the disk.

Reply:
Let's be sure to update the corresponding tests too.

Reply:
@arunsathiya I honestly don't remember, but I've added support for importing api keys, thanks!